The Sun C++ compiler reports errors building hufo-smoke:

"FirePart.h", line 125: Error: Overloading ambiguity between "std::pow(double, double)" and "std::pow(float, float)".
"FirePart.h", line 165: Error: Overloading ambiguity between "std::pow(double, double)" and "std::pow(float, float)".

Defining the FIREDA constant as a float instead of a double resolves this 
by making both arguments to the function be float, instead of one float
and one double.

Submitted upstream at:
https://sourceforge.net/tracker/?func=detail&aid=2767450&group_id=67131&atid=517001

diff -urp -x '*~' -x '*.orig' oglc_src/FirePart.h oglc_src/FirePart.h
--- oglc_src/FirePart.h	2005-04-09 10:50:18.000000000 -0700
+++ oglc_src/FirePart.h	2009-04-15 16:27:48.137837000 -0700
@@ -60,7 +60,7 @@ double nrnd (double d)		// normal distri
 #define FIRESIZE 1.0f
 #define FIREDS 0.7
 #define FIREALPHA 0.15f
-#define FIREDA 0.4
+#define FIREDA 0.4f
 #define FIRECYLR 0.2f
 
 struct Particle {
